src/main/java/seedu/todo/ui/controller/CommandController.java

```diff
@@ -0,0 +1,102 @@
+package seedu.todo.ui.controller;
+
+import seedu.todo.commons.util.StringUtil;
+import seedu.todo.logic.Logic;
+import seedu.todo.logic.commands.CommandResult;
+import seedu.todo.model.ErrorBag;
+import seedu.todo.ui.view.CommandErrorView;
+import seedu.todo.ui.view.CommandFeedbackView;
+import seedu.todo.ui.view.CommandInputView;
+
+/**
+ * Processes the input command from {@link CommandInputView}, pass it to {@link seedu.todo.logic.Logic}
+ * and hands the {@link seedu.todo.logic.commands.CommandResult}
+ * to {@link CommandFeedbackView} and {@link CommandErrorView}
+ * @author Wang Xien Dong
+ */
+public class CommandController {
+
+    private Logic logic;
+    private CommandInputView inputView;
+    private CommandFeedbackView feedbackView;
+    private CommandErrorView errorView;
+
+    /**
+     * Defines a default constructor
+     */
+    private CommandController() {}
```

Comment:
I disagree with the design of this class. I don't think the singleton pattern is appropriate here.